coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala

```diff
@@ -0,0 +1,219 @@
+          case Right((events, boundResetter)) =>
+            val iteratorOrError = try {
+              Right(JsonArrayIterator[JValue](events))
+            } catch { case _: JsonBadParse =>
+              Left(datasetBadRequest(BodyRequestError.NOT_JSON_ARRAY))
+            }
+            iteratorOrError match {
+              case Right(iterator) =>
+                val ProcessMutationReturns(copyNumber, version, lastModified, result) =
+                  processMutation(datasetId, iterator.map { ev => boundResetter(); mutateRate.mark(); ev }, tmp)
+                OK ~>
+                  ContentType(JsonContentType) ~>
+                  Header("X-SODA2-Truth-Last-Modified", dateTimeFormat.print(lastModified)) ~>
+                  Header("X-SODA2-Truth-Version", version.toString) ~>
+                  Header("X-SODA2-Truth-Copy-Number", copyNumber.toString) ~>
+                  Stream { w =>
+                    val bw = new BufferedOutputStream(w)
+                    bw.write('[')
+                    result.foreach(new Function1[MutationScriptCommandResult, Unit] {
+                      var didOne = false
+                      def apply(r: MutationScriptCommandResult) {
+                        if(didOne) bw.write(',')
+                        else didOne = true
+                        writeResult(bw, r, tmp)
+                      }
+                    })
+                    bw.write(']')
+                    bw.flush()
+
+                    log.debug("Non-linear index seeks: {}", tmp.stats.nonLinearIndexSeeks)
+                    log.debug("Non-linear data seeks: {}", tmp.stats.nonLinearDataSeeks)
+                  }
+              case Left(error) =>
+                error
+            }
+          case Left(response) =>
+            response
+        }
+      }
+      responseBuilder(resp)
+    }
+  }
+
+  def doDeleteDataset(req: HttpRequest): HttpResponse = {
+    deleteDataset(datasetId) match {
+      case DatasetDropper.Success =>
+        OK ~>
+          Header("X-SODA2-Truth-Last-Modified", dateTimeFormat.print(DateTime.now)) ~>
+          Header("X-SODA2-Truth-Copy-Number", 0.toString) ~>
+          Header("X-SODA2-Truth-Version", 0.toString) ~>
+          Content(JsonContentType, "[]")
+      case DatasetDropper.FailureNotFound =>
+        notFoundError(datasetId)
+      case DatasetDropper.FailureWriteLock =>
+        writeLockError(datasetId)
+    }
+  }
+
+
+
+  def doExportFile(req: HttpRequest): HttpResponse = {
+    val servReq = req.servletRequest
+    val precondition = req.precondition
+    val schemaHash = Option(servReq.getParameter("schemaHash"))
+    val ifModifiedSince = req.dateTimeHeader("If-Modified-Since")
+    val onlyColumns = Option(servReq.getParameterValues("c")).map(_.flatMap { c =>
+      norm(c).split(',').map(new UserColumnId(_)) }).map(UserColumnIdSet(_ : _*))
+    val limit = Option(servReq.getParameter("limit")).map { limStr =>
+      try {
```

Comment:
It's isomorphic to `Either[Throwable, A]` which is a terrifyingly vague type (`Throwable` is the next worst thing to a straight `AnyRef` in terms of precision of meaning).  What `Try` is for (my "never" before is hyperbole) is to allow vaguely try-catch looking code in a situation where you can't actually use try-catch because your control flow isn't block-oriented and so there is no dynamically-enclosing place that can know how to handle exceptions.  For example, if you're writing future-based code and so the thing dynamically enclosing you is just the future management runtime, not the code you'll ultimately be returning values for.

In particular, it's good to note that a signature like `A => Try[B]` doesn't promise that it won't throw exceptions (only that _some_ exceptions -- and it doesn't say which! -- will be returned as `Failure`s), so you have won neither safety by using it nor information about what kinds of failures can occur.